### PR TITLE
RFC counting, corrected RFC definitions

### PR DESCRIPTION
**Pull request by jrmyp**
These commits introduce RF filtering power computation, along with some cleanup and debug for the two definitions of RFC (Basic and MLast).

### CODE REVIEW

**Comment by kowey** [comment]
Looks like we have a small build failure here.  For some reason count_rfc seems to be still importing educe.stac.util.context (which has moved to educe.stac.context).

Also, it's generally better to avoid WIP style patches in a PR, but I understand that I haven't really taught you how to use `git rebase -i` yet

**Comment by jrmyp** [comment]
After merging from master, it should build correctly.

I made two entangled changes that I tried to split into separate commits, resulting in the WIP one, as the intermediate state wasn't in working order.

**Comment by kowey** [comment]
Hey @moreymat would you have any interest in reviewing this patch? Or would it be a bit plunging into the deep end at this point? It's not super-urgent.

**Comment by moreymat** [comment]
I could do it over the weekend or next week. If anyone want it done today,
you should do it.

**Comment by jrmyp** [comment]
Next week should be fine. The extra functionality is by no means urgent.

File: educe/stac/rfc.py [hunk 1]
```diff
@@ -52,24 +52,30 @@ class BasicRfc(object):
...
+        return self._build_frontier_from(last)
+
+    def _build_frontier_from(self, origin):
         """
         Given a dictionary mapping each node to its closest
-        right frontier node, generate a path up that frontier.
+        right frontier nodes and a start node, generate a path
+        up that frontier.
         """
         seen = set()
```
Comment (moreymat) [inline educe/stac/rfc.py]: If you can declare here `points = self._points`, then the code in the loop can be reverted to what it was, which is slightly more readable, plus

File: educe/stac/rfc.py [hunk 2]
```diff
@@ -138,21 +139,22 @@ def violations(self):
...
+            return list()
 
-        points = self._frontier_points(nodes)
-        nexts = itr.islice(nodes, 1, None)
-        for last, node1 in zip(nodes, nexts):
-            for lnk in graph.links(node1):
-                if not self._is_incoming_to(node1, lnk):
+        violations = list()
+        for i, new_node in enumerate(nodes):
+            last_node = nodes[i-1] if i>0 else None
```
Comment (moreymat) [inline educe/stac/rfc.py]: I don't have time to precisely work it out, but the semantics seems to have changed here.
I will go through it again next week.

**Comment by moreymat** [comment]
I have to stop for today, will resume reviewing this PR next week.

File: educe/stac/rfc.py (hunk 2 above)
Comment (moreymat) [inline educe/stac/rfc.py]: @kowey could I delegate the review of this portion of the code to you? I can't reconcile the intended semantics, the current version and the proposed new version.

File: educe/stac/tests.py [hunk 3]
```diff
@@ -378,6 +378,18 @@ def test_ambiguity_cdu_and_parent(self):
         self.assertNotIn(lg.get_edge('a', 'd'), violations)
         self.assertNotIn(lg.get_edge('b', 'd'), violations)
 
+    def test_first_incoming(self):
+        """ a -> b -> a (backwards link violation)"""
+        lg, graph = mk_graphs('#Aab / Saba')
```
Comment (moreymat) [inline educe/stac/tests.py]: Could you rename the EDUs in this example so the parallel with the second example immediately shows up?
Plus, these two tests could be merged into one test function (checking the same property on two different graphs).

Comment (jrmyp) [inline educe/stac/tests.py]: Done.

File: educe/stac/rfc.py (hunk 1 above)
Comment (jrmyp) [inline educe/stac/rfc.py]: Done.

**Pull request merged**